src/frontend/Axis.tcc

```diff
@@ -104,10 +108,18 @@ template <typename Backend> void Axis::draw_common_ticks(Backend &backend) {
     backend.move_to(vfloat2_t(m_pixels.bmin[0] - m_tick_len / 2, pos));
     backend.line_to(vfloat2_t(m_pixels.bmin[0], pos));
     std::snprintf(buffer, sizeof(buffer), "%.*g", m_sig_digits + 1, val);
+    
+    if (strlen(buffer) > max_char_len) {
+      max_char_len = strlen(buffer);
+    }
+
     backend.text(vfloat2_t(m_pixels.bmin[0] - m_tick_len / 2, pos), buffer,
                  NULL);
   }
 
+  /// TODO: Convertion from length * font_size to pixels
+  m_max_ytick_len = max_char_len * m_style.font_size() / 96.0 * 72.0;
```

Comment:
Where did 72/96 come from, just trial and error? Yes, agree that we need some way to get the exact pixel length of a string according to the current font, this should be a separate issue

Reply:
There is relation between font and pixels, i.e., for Roboto. It is siad that the scale factor is about 1.25. Anyway, now tests are merged with TestAxis.cpp. However, the current fix comes up with strange "exact pixel length". I am not aware the potential mechanism for the bugs yet.